src/components/table/OcTable.vue

```diff
@@ -216,8 +399,89 @@ export default {
     footerColspan() {
       return this.fields.length
     },
+    //my computed
+    groupedItems() {
+      let result = this.groupingFunction(
+        this.selected,
+        this.tableData,
+        this.groupingOrderAsc[this.selected]
+      )
+
+      if (this.groupingSettings && this.groupingSettings.previewTable && this.previewEnabled) {
+        let previewCount = this.groupingSettings.previewAmount + 1
+
+        result.forEach(e => {
+          let eLength = e.data.length
+          if (eLength - previewCount >= 0) e.data = e.data.slice(0, previewCount - 1)
+          previewCount = previewCount - eLength
+        })
+
+        result = result.filter(e => e.data.length > 0)
+      }
+
+      this.resultArray = result
+      return result
+    },
   },
   methods: {
+    //my methods
+    groupingOrder() {
+      let groupingOrder = {}
+      Object.keys(this.groupingSettings.groupingFunctions).forEach(
+        group => (groupingOrder[group] = true)
+      )
+      return groupingOrder
+    },
+    groupingFunction(col, data, asc) {
+      let result = {}
+
+      if (Object.keys(this.groupingSettings.groupingFunctions).includes(col)) {
+        data.forEach(row => {
+          result[this.groupingSettings.groupingFunctions[col](row)]
+            ? result[this.groupingSettings.groupingFunctions[col](row)].push(row)
+            : (result[this.groupingSettings.groupingFunctions[col](row)] = [row])
+        })
+        return this.produceGroups(Object.entries(result), asc, col)
+      }
+    },
+
+    produceGroups(arr, asc, col) {
+      let resultArray = []
+      for (const [key, value] of arr) {
```

Comment:
Why did you create a dict before, just to pass it here and create another dict? Why don't we merge this functionality in the calling function instead?

Reply:
Ups, I thought I had deleted this comment.. After re-reading I understood the reason and what I said was not true.